src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelEventGenerator.java

```diff
@@ -51,6 +63,40 @@ protected static String createRepoURI(String url, String projectName) {
         }
     }
 
+    protected static Link createLink(final String linkType, final String linkedEiffelEventType,
+            final File pluginDirectoryPath, final String projectName,
+            final String tableColumnName) {
+        try {
+        EventStorage eventStorage = EventStorageFactory.getEventStorage(pluginDirectoryPath, linkedEiffelEventType);
+        String lastEiffelEvent = getEiffelEventIdFromState(eventStorage, projectName, tableColumnName);
+
+        if (!StringUtils.isEmpty(lastEiffelEvent)) {
+            Link link = new Link();
+            link.type = linkType;
+            link.target = lastEiffelEvent;
+
+            return link;
+        }
+
+        } catch(IllegalArgumentException e) {
+            LOGGER.error("Failed creating link.", e);
+        }
+        return null;
+    }
+
+    protected static String getEiffelEventIdFromState(final EventStorage eventStorage, final String projectName,
```

Comment:
I think you need to change the name to `getEiffelEventIdFromStorage` or something

Reply:
Ye, i missed that one, thanks 😄